core/src/test/scala/com/yahoo/maha/core/query/QueryPipelineWithFallbackTest.scala

```diff
@@ -223,4 +223,99 @@ class QueryPipelineWithFallbackTest extends AnyFunSuite with Matchers with Befor
     pipeline = builder._1.toOption.get.build()
     assert(pipeline.fallbackQueryChainOption.isEmpty, s"No fallback query expected: $pipeline")
   }
+
+  private[this] def aLessThanBByLevelAndCostAndCardinality(a: (String, Engine, Long, Int, Int), b: (String, Engine, Long, Int, Int)): Boolean = {
```

Comment:
instead of replicating code here, we can expose public wrapper method to test this comparator in the DefaultQueryPipelineFactory.